### PR TITLE
[rmlui] Add option for LuaJIT bindings

### PR DESCRIPTION
**Pull request by max-dark**
Add option for LuaJIT bindings

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by max-dark** [comment]
@microsoft-github-policy-service agree

File: versions/r-/rmlui.json [hunk 1]
```diff
@@ -1,5 +1,20 @@
...
+    {
+      "git-tree": "551ed11d52a253a0a0165fa3397f0136ef95061d",
+      "version": "5.1-0",
+      "port-version": 0
+    },
+    {
+      "git-tree": "55710b58dd314006584812eafab1d89fc35815be",
+      "version": "5.1.0",
+      "port-version": 0
+    },
```
Comment (Cheney-W) [inline versions/r-/rmlui.json:17]:
```suggestion
    {
      "git-tree": "94de4eab8f339dd3c8887bbb728809685681245b",
      "version": "5.1",
      "port-version": 1
    },
```

**Review by Cheney-W — CHANGES_REQUESTED** [review]

**Comment by Cheney-W** [comment]
@max-dark Has the upstream approved your modifications?

**Comment by max-dark** [comment]
> @max-dark Has the upstream approved your modifications?

I don't know how to ask.

**Comment by Cheney-W** [comment]
You can submit an issue to [upstream](https://github.com/mikke89/RmlUi), explaining the changes you intend to make and the goals you hope to achieve, to see if they agree with your approach.

Also, one thing I noticed during local testing is that in your current modifications, when the `luajit` feature is enabled, the `find_package(LuaJIT REQUIRED)` is not being invoked. This is because after executing `set(BUILD_LUA_BINDINGS ON CACHE BOOL "Build Lua bindings")`, the value of `BUILD_LUA_BINDINGS` remains `OFF`. I think this might not align with what you expect.

**Pull request closed without merging**